### PR TITLE
 Add "speed" group to change the walking speed on and in blocks

### PR DESCRIPTION
**Pull request by rubenwardy**
https://github.com/minetest/minetest/pull/8613 but with devtest nodes

> As the name says, it allows you to easily change the speed of walking along various blocks without server load.
> This idea came to me from playerplus mod (https://notabug.org/TenPlus1/playerplus), which I have been using for a long time. Unfortunately, checking once every 1 second turned out to be insufficient, and checking 10 times a second I heavily loaded the server with many players. Therefore, it was decided to transfer this function to the engine. See the documentation.
> Just add the group "speed = -5" or "speed = 10" to any block and walk on it.
> Can be used to slow down in the snow, speeding up on asphalt, give the API modders, they will figure out what to do with it!

For #1822  and #7785

## To do

This PR is Ready for Review.

- [ ] Fix bug with standing position

## How to test

Place `testnodes:speed_slow_nonwalkable` and `testnodes:speed_slow_walkable` from devtest

### CODE REVIEW

**Comment by Wuzzy2** [comment]
Should also be tested in combination with `move_resistance`.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1837,6 +1837,9 @@ to games.
 
 * `disable_repair`: If set to 1 for a tool, it cannot be repaired using the
   `"toolrepair"` crafting recipe
+* `speed`: Changes movement speed for players and items will moving on the node.
+	Only a decrease in speed is possible.
+   * `speed = -1`: to slow down the movement (min. `-100`, unable to walk)
```
Comment (nerzhul) [inline doc/lua_api.txt:1842]: i like the feature but i'd prefer if speed is a percentage, 100% = normal 0% = unable to walk 1000% = 10x the normal speed. It's better in terms of understanding that having complicated calculation.  speedFactor = uint16 and that's all, no divide, no addition, no clamp, just a raw value factor value

**Review by nerzhul — CHANGES_REQUESTED** [review]

**Comment by Wuzzy2** [comment]
@nerzhul: This will break walking because `speed = 0` is the implied default value, causing all players to come to a screeching halt. It's the same reason why we have `fall_damage_add_percent`.

**Comment by Wuzzy2** [comment]
OK, so it seems to interact fine in combination with `move_resistance` (although I think this combination will be rare, but still!).

I noticed that `speed` also affects climbing speed. I’m not sure if I like that. If we ever add a modifier to modify climb speed only, there will be an annoying conflict. Best to leave climb speed unchanged by `speed` and only touch the walking speed. I think walk speed and climb speed should be changed with two independent properties. There’s a PR for climb speed: #11376.

----

The test nodes need improvement. First, the name. I suggest more generic name for consistency with everything else. I suggest: "Slow non-walkable block (-50%)", "Slow block (-50%)".

Second, the texture seems invalid. It has transparent pixels but is rendered for a `normal` node. I know this is only DevTest but that's no excuse to use invalid data for the test/example nodes.

Third, `disable_jump=1` should be removed. Test nodes in DevTest are meant to test *one* feature *in isolation of all other relevant features*. A combination of features is only OK if it's explicitly stated as such.

----

New, improved DevTest code:

```
for i=25,100,25 do
	local tile = "testnodes_speed_slow.png"
	if i == 100 then
		tile = "testnodes_speed_stop.png"
	else
		tile = tile .. "^[colorize:#000000:"..(i*2)
	end

	-- Slow block, nonwalkable
	minetest.register_node("testnodes:speed_slow_nonwalkable_"..i, {
		description = S("Slow non-walkable Block (-@1%)", i),
		groups = {dig_immediate = 3, speed = -i},
		walkable = false,
		climbable = true,

		drawtype = "plantlike",
		tiles = {tile},
		paramtype = "light",
		sunlight_propagates = true,
	})

	-- Slow block, walkable
	minetest.register_node("testnodes:speed_slow_walkable_"..i, {
		description = S("Slow Block (-@1%)", i),
		groups = {dig_immediate = 3, speed = -i},


		tiles = {tile},
	})
end
```

Textures:

* https://satoshiupload.com/images/4qBmsgVr6I.png
* https://satoshiupload.com/images/1NG6AQ74s9.png

**Comment by Wuzzy2** [comment]
Correction: From the previous comment, delete `climbable = true,`.

File: doc/lua_api.txt (hunk 1 above)
Comment (SmallJoker) [reply doc/lua_api.txt:1842]: Unspecified groups default to 0, hence it would require special treatment for a specific group.

Comment (appgurueu) [reply doc/lua_api.txt:1842]: Why should only a decrease be possible? Sounds arbitrary to me.

**Comment by appgurueu** [comment]
> checking 10 times a second I heavily loaded the server with many players

Note that this claim is pretty much wrong. The main reason for this would not be reducing server load, but rather improving client prediction in order to keep movement smooth even if lag is high.

**Comment by BluebirdGreycoat** [comment]
I don't believe this should be abandoned, the concept is good. I've had a script that does basically this for quite a long time, but obviously it does not work well when lag is high.

**Comment by rubenwardy** [comment]
It's supported, just needs adoption

**Pull request closed without merging**